src/components/KYCWall/BaseKYCWall.js

```diff
@@ -93,10 +95,19 @@ class KYCWall extends React.Component {
      */
     selectPaymentMethod(paymentMethod) {
         this.props.onSelectPaymentMethod(paymentMethod);
-        if (paymentMethod === CONST.PAYMENT_METHODS.BANK_ACCOUNT) {
+        if (paymentMethod === CONST.PAYMENT_METHODS.PERSONAL_BANK_ACCOUNT) {
             Navigation.navigate(this.props.addBankAccountRoute);
         } else if (paymentMethod === CONST.PAYMENT_METHODS.DEBIT_CARD) {
             Navigation.navigate(this.props.addDebitCardRoute);
+        } else if (paymentMethod === CONST.PAYMENT_METHODS.BUSINESS_BANK_ACCOUNT) {
+            if (ReportUtils.isIOUReport(this.props.iouReport)) {
+                const policyID = Policy.createWorkspaceFromIOUPayment(this.props.iouReport);
+
+                // Navigate to the bank account set up flow for this specific policy
+                Navigation.navigate(ROUTES.BANK_ACCOUNT_WITH_STEP_TO_OPEN.getRoute('', policyID));
+                return;
+            }
+            Navigation.navigate(this.props.addBankAccountRoute);
```

Comment:
I think this default might be incorrect. I think this defaults to a personal bank account since we pass it [here](https://github.com/Expensify/App/blob/7a68a3d1003130e63065bcbef253c912a3431c08/src/pages/settings/Wallet/WalletPage/WalletPage.js#L381)

Reply:
I think this should be correct because we pass it to the SettlementButton in both cases using this method which checks if the parent report is workspace chat or DM https://github.com/Expensify/App/blob/e909316f451a08aedd420c31f19cc3ea3cac0243/src/components/MoneyReportHeader.js#L94